src/main/java/edu/harvard/iq/dataverse/DatasetServiceBean.java

```diff
@@ -1042,4 +1061,492 @@ public void deleteHarvestedDataset(Dataset dataset, DataverseRequest request, Lo
+
+        List<CompletableFuture<fileDetailsHolder>> hashvalueCompletableFutures =
+                inputList.stream().map(iD -> calculateDetailsAsync(iD,globusLogger)).collect(Collectors.toList());
+
+        CompletableFuture<Void> allFutures = CompletableFuture
+                .allOf(hashvalueCompletableFutures.toArray(new CompletableFuture[hashvalueCompletableFutures.size()]));
+
+        CompletableFuture<List<fileDetailsHolder>> allCompletableFuture = allFutures.thenApply(future -> {
+            return hashvalueCompletableFutures.stream()
+                    .map(completableFuture -> completableFuture.join())
+                    .collect(Collectors.toList());
+        });
+
+        CompletableFuture completableFuture = allCompletableFuture.thenApply(files -> {
+            return files.stream().map(d -> json(d)).collect(toJsonArray());
+        });
+
+        JsonArrayBuilder filesObject = (JsonArrayBuilder) completableFuture.get();
+
+        JsonObject output = Json.createObjectBuilder().add("files", filesObject).build();
+
+        return output;
+
+    }
+
+    private CompletableFuture<fileDetailsHolder> calculateDetailsAsync(String id, Logger globusLogger) {
+        //logger.info(" calcualte additional details for these globus id  ==== " + id);
+
+        return CompletableFuture.supplyAsync( () -> {
+            try {
+                Thread.sleep(2000);
+            } catch (InterruptedException e) {
+                e.printStackTrace();
+            }
+            try {
+                return ( calculateDetails(id,globusLogger) );
+            } catch (InterruptedException | IOException e) {
+                e.printStackTrace();
+            }
+            return null;
+        }, executor).exceptionally(ex -> {
+            return null;
+        });
+    }
+
+
+    private fileDetailsHolder calculateDetails(String id, Logger globusLogger) throws InterruptedException, IOException {
+        int count = 0;
+        String checksumVal = "";
+        InputStream in = null;
+        String fileId = id.split("IDsplit")[0];
+        String fullPath = id.split("IDsplit")[1];
+        String fileName = id.split("IDsplit")[2];
+
+        // what if the file doesnot exists in s3
+        // what if checksum calculation failed
+
+        do {
+            try {
+                StorageIO<DvObject> dataFileStorageIO = DataAccess.getDirectStorageIO(fullPath);
+                in = dataFileStorageIO.getInputStream();
+                checksumVal = FileUtil.calculateChecksum(in, DataFile.ChecksumType.MD5);
+                count = 3;
+            }catch (IOException ioex) {
+                count = 3;
+                logger.info(ioex.getMessage());
+                globusLogger.info("S3AccessIO: DataFile (fullPAth " + fullPath + ") does not appear to be an S3 object associated with driver: " );
+            }catch (Exception ex) {
```

Comment:
What exceptions get caught here? Just curious as to why they are things you want to retry on whereas IOExceptions aren't.